### ISSUE TITLE
Nuitka does not work when using the mitmproxy module in multiprocessing but works well when I use Pyinstaller

### ISSUE THREAD

**Issue by Pragmatism0220** [Nuitka/Nuitka#1950]
* Nuitka version, full Python version, flavor, OS, etc. as output by *this*
  command.

```shell
1.2.7
Commercial: None
Python: 3.10.8 | packaged by conda-forge | (main, Nov 22 2022, 08:16:33) [MSC v.1929 64 bit (AMD64)]
Flavor: Anaconda Python
Executable: D:\miniconda3\envs\mortal\python.exe
OS: Windows
Arch: x86_64
WindowsRelease: 10
```

* How did you install Nuitka and Python

  pip in miniconda

* The specific PyPI names and versions

```shell
absl-py==1.3.0
altgraph==0.17.3
asgiref==3.3.4
blinker==1.4
Brotli==1.0.9
cachetools==5.2.0
certifi==2022.9.24
cffi @ file:///D:/bld/cffi_1671179506518/work
charset-normalizer==2.1.1
click==7.1.2
colorama==0.4.6
comtypes==1.1.14
cryptography==3.3.2
docopt==0.6.2
fake-useragent==1.1.1
Flask==1.1.4
future==0.18.2
google-auth==2.14.1
google-auth-oauthlib==0.4.6
grpcio==1.50.0
h11==0.14.0
h2==4.1.0
hpack==4.0.0
hyperframe==6.0.1
idna==3.4
itsdangerous==1.1.0
Jinja2==2.11.3
kaitaistruct==0.9
ldap3==2.8.1
Markdown==3.4.1
MarkupSafe==1.1.1
mitmproxy==6.0.2
msgpack==1.0.4
Nuitka==1.2.7
numpy==1.23.5
oauthlib==3.2.2
ordered-set==4.1.0
passlib==1.7.4
pefile==2022.5.30
Pillow==9.3.0
pipreqs==0.4.11
protobuf==3.14.0
psutil==5.9.4
publicsuffix2==2.20191221
pyasn1==0.4.8
pyasn1-modules==0.2.8
pycparser @ file:///home/conda/feedstock_root/build_artifacts/pycparser_1636257122734/work
pydivert==2.1.0
pyinstaller==5.6.2
pyinstaller-hooks-contrib==2022.13
pyOpenSSL==20.0.1
pyparsing==2.4.7
pyperclip==1.8.2
pypiwin32==223
pyttsx3==2.90
pywin32==305
pywin32-ctypes==0.2.0
requests==2.28.1
requests-oauthlib==1.3.1
rsa==4.9
ruamel.yaml==0.16.13
six==1.16.0
sortedcontainers==2.3.0
tensorboard==2.11.0
tensorboard-data-server==0.6.1
tensorboard-plugin-wit==1.8.1
tinyaes==1.0.4
toml==0.10.2
torch==1.13.0
tornado==6.2
tqdm==4.64.1
typing_extensions==4.4.0
urllib3==1.26.13
urwid==2.1.2
Werkzeug==1.0.1
wsproto==1.0.0
yarg==0.1.9
zstandard==0.19.0
```

* Also supply a [Short, Self Contained, Correct, Example](https://sscce.org/)

  The project struture is here, very simple, only two files:
```shell
└── wrapper
│   └── addons.py
└── main.py
```
The code is here. First is `main.py` and second is `addons.py`:
```py
# main.py
# -*- coding: utf-8 -*-
import os
import sys
import time
from multiprocessing import freeze_support, Process, Manager


def start_mitmproxy(port, mitm_flag):
    sys.stdout = open(os.devnull, 'w')
    sys.stderr = open(os.devnull, 'w')
    mitm_flag['start'] = True

    from wrapper.addons import start_mitmproxy as sm
    sm(port)


if __name__ == '__main__':
    freeze_support()

    manager = Manager()
    mitmproxy_flag = manager.dict({'start': False})
    mitmproxy_process = Process(target=start_mitmproxy, args=(18080, mitmproxy_flag,))
    mitmproxy_process.daemon = True
    mitmproxy_process.start()
    while True:
        if mitmproxy_flag['start']:
            break
        time.sleep(1)

    os.system('pause')
```
`addons.py` is here:
```py
# addons.py
# -*- coding: utf-8 -*-
import pickle
import threading
from xmlrpc.server import SimpleXMLRPCServer

import mitmproxy.http
import mitmproxy.websocket
from mitmproxy.options import Options
from mitmproxy.proxy.config import ProxyConfig
from mitmproxy.proxy.server import ProxyServer
from mitmproxy.tools.dump import DumpMaster


activated_flows = []  # store all flow.id ([-1] is the recently opened)
messages_dict = dict()  # flow.id -> List[flow_msg]

RPC_server = None
Master = None


class ClientWebSocket:
    def __init__(self, show=False):
        self.show = show  # Whether to print information to the console. (Default: False)
        pass

    # Websocket lifecycle
    def websocket_handshake(self, flow: mitmproxy.http.HTTPFlow):
        """
            Called when a client wants to establish a WebSocket connection. The
            WebSocket-specific headers can be manipulated to alter the
            handshake. The flow object is guaranteed to have a non-None request
            attribute.
        """
        if self.show:
            print('[handshake websocket]:', flow, flow.__dict__, dir(flow))

    def websocket_start(self, flow: mitmproxy.websocket.WebSocketFlow):
        """
            A websocket connection has commenced.
        """
        if self.show:
            print('[new websocket]:', flow, flow.__dict__, dir(flow))
        global activated_flows, messages_dict
        activated_flows.append(flow.id)
        messages_dict[flow.id] = flow.messages

    def websocket_message(self, flow: mitmproxy.websocket.WebSocketFlow):
        """
            Called when a WebSocket message is received from the client or
            server. The most recent message will be flow.messages[-1]. The
            message is user-modifiable. Currently there are two types of
            messages, corresponding to the BINARY and TEXT frame types.
        """
        flow_msg = flow.messages[-1]

        packet = flow_msg.content
        from_client = flow_msg.from_client
        if self.show:
            print("[" + ("Sended" if from_client else "Reveived") +
                  "] from '" + flow.id + "': decode the packet here: %r…" % packet)

    def websocket_error(self, flow: mitmproxy.websocket.WebSocketFlow):
        """
            A websocket connection has had an error.
        """
        if self.show:
            print("websocket_error, %r" % flow)

    def websocket_end(self, flow: mitmproxy.websocket.WebSocketFlow):
        """
            A websocket connection has ended.
        """
        if self.show:
            print('[end websocket]:', flow, flow.__dict__, dir(flow))
        global activated_flows, messages_dict
        activated_flows.remove(flow.id)
        messages_dict.pop(flow.id)


# RPC Call Function
def get_len() -> int:
    global activated_flows, messages_dict
    L = messages_dict[activated_flows[-1]]
    return len(L)


def get_item(id: int):
    global activated_flows, messages_dict
    L = messages_dict[activated_flows[-1]]
    return pickle.dumps(L[id])


def get_items(from_: int, to_: int):
    global activated_flows, messages_dict
    L = messages_dict[activated_flows[-1]]
    return pickle.dumps(L[from_:to_:])


def RPC_init():
    server = SimpleXMLRPCServer(('localhost', 37247))
    server.register_function(get_len, "get_len")
    server.register_function(get_item, "get_item")
    server.register_function(get_items, "get_items")
    # print("RPC Server Listening on 127.0.0.1:37247 for Client.")
    server.serve_forever()


class ProxyMaster(DumpMaster):
    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs)

    def run(self, func=None):
        try:
            DumpMaster.run(self)
        except KeyboardInterrupt:
            self.shutdown()


def start_mitmproxy(port):
    global RPC_server, Master
    RPC_server = threading.Thread(target=RPC_init)
    RPC_server.start()
    options = Options(listen_host='127.0.0.1', listen_port=int(port), ssl_insecure=True, upstream_cert=False)
    config = ProxyConfig(options)
    Master = ProxyMaster(options, with_termlog=False, with_dumper=False)
    Master.server = ProxyServer(config)
    Master.addons.add(ClientWebSocket())
    Master.run()
```

* Provide in your issue the Nuitka options used

  command:
```shell
nuitka --standalone --onefile --remove-output main.py
```

Nuitka can package correctly to generate an exe file. However, the runtime will find that mitmproxy is not started correctly as expected. The same code can run correctly with the latest version of Pyinstaller without any modification.

The expected result should be that the mitmproxy successfully proxies and listens to the data on the local port 18080.

This really breaks me down.

**Comment by Pragmatism0220** [comment]
Okay, I think I know why. Nuitka did not detect that several modules under the passlib.handlers package were referenced in the mitmproxy module during packaging, whereas Pyinstaller did. To be safe, I just explicitly add the `--include-package=passlib.handlers` option and it works well.

**Comment by kayhayen** [comment]
Using Anaconda and pip, I couldn't get it past "from mitmproxy.proxy.config import ProxyConfig", that seems not found despite mitproxy being installed, seems your code is specific to the 6 version you gave. Also it passed for me, because you report start in your reproducer too soon, so it ends before anything bad happens.

**Comment by kayhayen** [comment]
The reproducer is bad, but I will give it a try in a virtualenv, for now a quick test wasn't possible, that doesn't mean it will not be done. I am curious to what module has that hidden dependency.

**Comment by Pragmatism0220** [comment]
> The reproducer is bad, but I will give it a try in a virtualenv, for now a quick test wasn't possible, that doesn't mean it will not be done. I am curious to what module has that hidden dependency.

Thank you for your reply! According to my observation, in the version of `mitmproxy==6.0.2`, `passlib.apache` is referenced, where `passlib.handler.bcrypt` is not explicitly referenced in `passlib`. This may be the cause of the problem. The details are as follows:
![image](https://user-images.githubusercontent.com/55100969/209138903-428a17f4-8a25-4d28-a27a-70910e187ec9.png)
The structure of the `passlib.handler` package is as follows:
![image](https://user-images.githubusercontent.com/55100969/209139044-75e84054-a12a-4051-83ed-a33296f1f92c.png)

**Comment by kayhayen** [comment]
Ok, I think I can use that to reproduce it easily, thanks!

**Comment by kayhayen** [comment]
This is the user yaml configuration that should do, I will include it in 1.3.2 right now:

```
- module-name: 'passlib.apache'
  implicit-imports:
    - depends:
        - 'passlib.handlers.digests'
        - 'passlib.handlers.bcrypt'
        - 'passlib.handlers.md5_crypt'
        - 'passlib.handlers.des_crypt'
        - 'passlib.handlers.ldap_digests'

```

**Comment by kayhayen** [comment]
This was released as part of 1.3.2 hotfix release.
